<launch>

  <node pkg="joy" type="joy_node" name="joy_node" respawn="true" >
	  <param name="autorepeat_rate" value="20.0" />
	  <param name="dev" type="string" value="/dev/input/js0" />
	  <param name="deadzone" value="0.12" />
  </node>

  <node pkg="wheelchair_base_controller" type="joy2chair.py" name="joy2chair" output="screen" respawn="true" >
	  <param name="port_name" value="/dev/sensors/ftdi_A600aivC" />
	  <remap from="joy" to="/joy_reflex" />
  </node>

  <node pkg="wheelchair_base_controller" type="joy2twist.py" name="joytwisterizer" respawn="true" />

  <node pkg="wheelchair_base_controller" type="otto_ekf.py" name="Otto_EKF" respawn="true" />
  
  <node pkg="reflexive_avoidance" type="reflexive_obstacle_avoidance" name="reflexive_halt" output="screen" respawn="true" >
    <remap from="scan" to="/neato/scan_final" />
    <remap from="range" to="/sonars" />
    <remap from="input_cloud" to="/camera/depth/points_collision" />
    <remap from="joy" to="/joy" />
    <remap from="joy_output" to="/joy_reflex" />
    <rosparam>
      use_kinect: true
      num_points: 100
      use_sonars: true
      front_sonar_thresh: 0.35
      side_sonar_thresh: 0.3
      use_laser: true
      front_laser_thresh: 0.5
      front_laser_angle: 2.0
      side_laser_thresh: 0.45
      side_laser_angle: 2.0
    </rosparam>
  </node>
  
</launch>